src/unstract/sdk/file_storage/fs_impl.py

```diff
@@ -51,6 +52,8 @@ def read(
                 if seek_position > 0:
                     file_handle.seek(seek_position)
                 return file_handle.read(length)
+        except FileNotFoundError as e:
+            raise e
```

Comment:
@gaya3-zipstack ideally this is not even required since we are catching and re-raising the same exception. I assume we had to add this to avoid incorrectly marking `FileNotFoundError` as `FileOperationError` when we do a catch all of `except Exception`.

I suggest moving away from this catch all if it does not help our error handling. If you think that wrapping all exceptions with our own class is required, I suggest the below refactoring
```
except Exception as e:
    if isinstance(e, FileNotFoundError):
        raise e
    raise FileOperationError(str(e)) from e
```

Reply:
@chandrasekharan-zipstack  I will refactor this in the nexr PR. However at this point, we are trying to exercise flows for the dpendent services. The reason of catchin FileNOtFound explicitly is because on callers, in some places, looks like we are dependent on this for certain assumptions. To facilitate that is why we are doing this.